app/services/atmosphere/billing_service.rb

```diff
@@ -131,12 +140,25 @@ def self.can_afford_vm?(appliance, vm)
       if appliance.fund.blank?
         raise Atmosphere::BillingException.new(message: "can_afford_vm? invoked on an appliance (with id #{appliance.id}) which has no fund assigned. Unable to proceed.")
       end
-      billable_time = self.calculate_billable_time(appliance, Time.now.utc, true)
-      amt_due = (billable_time*(vm.virtual_machine_flavor.hourly_cost)/(vm.appliances.count+1)).round
+
+      # Check whether the target VM is already assigned to the appliance via deployments
+      # If so, respect the prepayment period which *may* be present for this deployment
+      deployments = appliance.deployments.select {|dep| dep.virtual_machine == vm}
```

Comment:
```ruby
deployments = appliance.deployments.where(virtual_machine_id: vm.id)
```

BTW I believe that it is not possible to have more than 1 deployment relation between appliance and vm, then we can use:

```ruby
deployments = appliance.deployments.find_by(virtual_machine_id: vm.id)
```